sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/ServiceEndpointTest.java

```diff
@@ -0,0 +1,75 @@
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.AsyncDocumentClient;
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.TestConfigurations;
+import com.azure.cosmos.models.CosmosItemResponse;
+import com.azure.cosmos.rx.TestSuiteBase;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ServiceEndpointTest extends TestSuiteBase {
+
+    @DataProvider
+    private Object[][] serviceEndpointArgProvider() {
+        return new Object[][] {
+            { TestConfigurations.HOST.substring(0,TestConfigurations.HOST.length() - 1) },
+            { TestConfigurations.HOST },
+            { TestConfigurations.HOST.replace("https://", "") },
+            { TestConfigurations.HOST.replace("https://", "").replace("/", "") },
+            { TestConfigurations.HOST + ";AccountKey=" + TestConfigurations.MASTER_KEY },
+            { TestConfigurations.HOST + ";" + TestConfigurations.MASTER_KEY + ";" },
+            { TestConfigurations.HOST + TestConfigurations.MASTER_KEY + ";" },
+            { TestConfigurations.HOST + TestConfigurations.MASTER_KEY },
+            { null },
+            { "" }
+        };
+    }
+
+    @Test(groups = { "unit" }, dataProvider = "serviceEndpointArgProvider")
+    public void validateServiceEndpoint(String serviceEndpoint) {
+        CosmosClient cosmosClient;
+        try {
+            CosmosClientBuilder cosmosClientBuilder = new CosmosClientBuilder()
+                .endpoint(serviceEndpoint)
+                .key(TestConfigurations.MASTER_KEY);
+            cosmosClient = cosmosClientBuilder.buildClient();
+        } catch (NullPointerException e) {
+            assertThat(serviceEndpoint).isNull();
```

Comment:
Should we ever be expecting a NPE?

Reply:
The existing behavior is that we throw a null pointer exception when null is passed in for the service endpoint.  There are tests in spring data cosmos validating this.